editor/src/components/canvas/floating-panels.tsx

```diff
@@ -0,0 +1,754 @@
+            droppedToOutsideOfAColumn.targetPanel === 'rightMenu2' &&
+            rightColumn2.length > 0 &&
+            (rightColumn.length === 0 ||
+              (draggedPanel === 'rightMenu1' && rightColumn.length === 1))
+          ) {
+            // dragging to far right when something is already there
+            shouldSwitchPanel = 'rightMenu1'
+          }
+
+          return {
+            newPanel: droppedToOutsideOfAColumn.targetPanel,
+            switchWithPanel: shouldSwitchPanel,
+          }
+        }
+
+        return {
+          newPanel:
+            panelsData.find((panel) => {
+              if (panel.frame == null) {
+                return null
+              }
+              return rectContainsPoint(panel.frame, newPosition)
+            })?.location ?? null,
+          switchWithPanel: null,
+        }
+      })()
+
+      const currentPanelData = panelsData.find((v) => v.name === draggedMenuOrPane)
+      if (
+        currentPanelData != null &&
+        newPanel != null &&
+        (draggedPanel != newPanel || switchWithPanel != null)
+      ) {
+        const columnData = panelsData.filter((d) => d.location === newPanel)
+        let updatedPanelData = panelsData.map((panel) => {
+          if (panel.name === draggedMenuOrPane) {
+            return {
+              ...panel,
+              location: newPanel,
+              locationInColumn: columnData.length,
+            }
+          } else if (panel.location === newPanel && switchWithPanel != null) {
+            return { ...panel, location: switchWithPanel }
+          } else if (draggedPanel === panel.location) {
+            return {
+              ...panel,
+              locationInColumn:
+                panel.locationInColumn > currentPanelData.locationInColumn
+                  ? panel.locationInColumn - 1
+                  : panel.locationInColumn,
+            }
+          } else {
+            return panel
+          }
+        })
+
+        // Readjust the locationInColumn values to have them start at zero and increment by 1.
+        updatedPanelData.sort(
+          (firstPanel, secondPanel) => firstPanel.locationInColumn - secondPanel.locationInColumn,
+        )
+        let locationInColumnMap: Map<PanelName, number> = new Map()
+        updatedPanelData = updatedPanelData.map((panelData) => {
+          const currentLocationInColumnValue = locationInColumnMap.get(panelData.location)
+          const newLocationInColumn =
+            currentLocationInColumnValue == null ? 0 : currentLocationInColumnValue + 1
+          locationInColumnMap.set(panelData.location, newLocationInColumn)
+          if (panelData.locationInColumn === newLocationInColumn) {
+            return panelData
+          } else {
+            return {
+              ...panelData,
+              locationInColumn: newLocationInColumn,
+            }
+          }
+        })
+
+        // Put the entries in the same order by name.
+        updatedPanelData = [
```

Comment:
Even though the various components appeared to have the correct `key` properties, this fixed an issue where the code editor would reload when being repositioned to the left of or below another part.